lib/Model.php

```diff
@@ -456,8 +456,16 @@ public function assign_attribute($name, $value)
 		if ($value instanceof DateTime)
 			$value->attribute_of($this,$name);
 
+		if (array_key_exists($name,$this->attributes)) {
+			if ($this->attributes[$name] != $value) {
```

Comment:
ok som si to pozornejsie precital :) , matuco je to napisane. Pre jednoduchost je nalepsie otocit celu logiku:
```php
if (!array_key_exists($name, $this->attributes) OR array_key_exists($name, $this->attributes) AND $this->attributes[$name] != $value) {
     $this->flag_dirty($name);
}

// alebo este troska bez opakovania
$keyExists = array_key_exists($name, $this->attributes);  // premenovat premenu na to co dava zmysel v contexte napr ($valueChange, $isNew)
if (!$keyExists OR $keyExists AND $this->attributes[$name] != $value) {
     $this->flag_dirty($name);
}
```